azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -460,6 +968,256 @@ func FlattenDefaultNodePool(input *[]containerservice.ManagedClusterAgentPoolPro
+	if input == nil {
+		return []interface{}{}
+	}
+
+	var cpuManagerPolicy, cpuCfsQuotaPeriod, topologyManagerPolicy string
+	var cpuCfsQuotaEnabled bool
+	var imageGcHighThreshold, imageGcLowThreshold, containerLogMaxSizeMB, containerLogMaxLines, podMaxPids int
+
+	if input.CPUManagerPolicy != nil {
+		cpuManagerPolicy = *input.CPUManagerPolicy
+	}
+	if input.CPUCfsQuota != nil {
+		cpuCfsQuotaEnabled = *input.CPUCfsQuota
+	}
+	if input.CPUCfsQuotaPeriod != nil {
+		cpuCfsQuotaPeriod = *input.CPUCfsQuotaPeriod
+	}
+	if input.ImageGcHighThreshold != nil {
+		imageGcHighThreshold = int(*input.ImageGcHighThreshold)
+	}
+	if input.ImageGcLowThreshold != nil {
+		imageGcLowThreshold = int(*input.ImageGcLowThreshold)
+	}
+	if input.TopologyManagerPolicy != nil {
+		topologyManagerPolicy = *input.TopologyManagerPolicy
+	}
+	if input.ContainerLogMaxSizeMB != nil {
+		containerLogMaxSizeMB = int(*input.ContainerLogMaxSizeMB)
+	}
+	if input.ContainerLogMaxFiles != nil {
+		containerLogMaxLines = int(*input.ContainerLogMaxFiles)
+	}
+	if input.PodMaxPids != nil {
+		podMaxPids = int(*input.PodMaxPids)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"cpu_manager_policy":        cpuManagerPolicy,
+			"cpu_cfs_quota_enabled":     cpuCfsQuotaEnabled,
+			"cpu_cfs_quota_period":      cpuCfsQuotaPeriod,
+			"image_gc_high_threshold":   imageGcHighThreshold,
+			"image_gc_low_threshold":    imageGcLowThreshold,
+			"topology_manager_policy":   topologyManagerPolicy,
+			"allowed_unsafe_sysctls":    utils.FlattenStringSlice(input.AllowedUnsafeSysctls),
+			"container_log_max_size_mb": containerLogMaxSizeMB,
+			"container_log_max_line":    containerLogMaxLines,
+			"pod_max_pid":               podMaxPids,
+		},
+	}
+}
+
+func flattenAgentPoolLinuxOSConfig(input *containerservice.LinuxOSConfig) ([]interface{}, error) {
+	if input == nil {
+		return make([]interface{}, 0), nil
+	}
+
+	var swapFileSizeMB int32
+	if input.SwapFileSizeMB != nil {
+		swapFileSizeMB = *input.SwapFileSizeMB
+	}
+	var transparentHugePageDefrag string
+	if input.TransparentHugePageDefrag != nil {
+		transparentHugePageDefrag = *input.TransparentHugePageDefrag
+	}
+	var transparentHugePageEnabled string
+	if input.TransparentHugePageEnabled != nil {
+		transparentHugePageEnabled = *input.TransparentHugePageEnabled
+	}
+	sysctlConfig, err := flattenAgentPoolSysctlConfig(input.Sysctls)
+	if err != nil {
+		return nil, err
+	}
+	return []interface{}{
+		map[string]interface{}{
+			"swap_file_size_mb":             swapFileSizeMB,
+			"sysctl_config":                 sysctlConfig,
+			"transparent_huge_page_defrag":  transparentHugePageDefrag,
+			"transparent_huge_page_enabled": transparentHugePageEnabled,
+		},
+	}, nil
+}
+
+func flattenAgentPoolSysctlConfig(input *containerservice.SysctlConfig) ([]interface{}, error) {
+	if input == nil {
+		return make([]interface{}, 0), nil
+	}
+
+	var fsAioMaxNr int32
```

Comment:
this should be an int, rather than an int32, since that's not a valid within the context of `d.Get/Set` (same for all of these)